### PR TITLE
Updating the docs to cater for creating workspaces

### PR DESCRIPTION
**Pull request by ross-p-smith**
# PR for issue #1232 

## What is being addressed
- Updated Auth Doc
- Updated Getting started doc wrt Auth
- Update workspace doc wrt Auth

### CODE REVIEW

File: docs/tre-admins/setup-instructions/installing-base-workspace.md [hunk 1]
```diff
@@ -33,13 +33,18 @@ Now that we have published and registered a base workspace bundle we can use the
...
+    --swaggerui-clientid <SWAGGER_UI_CLIENT_ID> \
+    --admin-consent \
+    --automation-clientid <AUTOMATION_ADMIN_ACCOUNT_CLIENT_ID>
 ```
 
 !!! caution
     If you're using a separate tenant for AAD app registrations to the one where you've deployed the TRE infrastructure resources, ensure you've signed into that tenant in the `az cli` before running the above command. See **Using a separate Azure Active Directory tenant** in [Pre-deployment steps](./pre-deployment-steps.md) for more details.
 
-Running the script will report `WORKSPACE_API_CLIENT_ID` for the generated app which needs to be used in the POST body below.
+Running the script will report `WORKSPACE_API_CLIENT_ID` and `WORKSPACE_API_CLIENT_SECRET` for the generated app. It is a good idea to copy these into `/templates/core/.env`. You need to use `WORKSPACE_API_CLIENT_ID` in the POST body below.
```
Comment (daltskin) [inline docs/tre-admins/setup-instructions/installing-base-workspace.md:47]: nit: Suggest making this a mandatory step eg. something like "Running the script will return... copy these into the .env file for later reference"

File: docs/tre-admins/setup-instructions/pre-deployment-steps.md [hunk 2]
```diff
@@ -60,14 +60,18 @@ Next, you will set the configuration variables for the specific Azure TRE instan
...
   ```
 
   !!! note
-      The full functionality of the script requires directory admin privileges. You may need to contact your friendly Azure Active Directory admin to complete this step. The app registrations can be created manually in Azure Portal too. For more information, see [Authentication and authorization](../auth.md).
+      The full functionality of the script requires directory admin privileges. You may need to contact your friendly Azure Active Directory admin to complete this step. The app registrations can be created manually in Azure Portal too. For more information, see [Authentication and authorization](../auth.md). You can run the script without the `--admin-consent` and ask your admin to grant consent.
   
   !!! note
       If you don't have permissions and just want to create a development environment then skip this step and see the steps in the "Using a separate Azure Active Directory tenant) below.
 
+  !!! note
```
Comment (daltskin) [inline docs/tre-admins/setup-instructions/pre-deployment-steps.md:72]: nit: We now have 3 "note" sections here, could we consolidate these at all?

**Review by daltskin — COMMENTED** [review]
Made some suggestions

File: docs/tre-admins/auth.md [hunk 3]
```diff
@@ -1,55 +1,87 @@
...
+| Workspace API | Typically you would have an application securing one or more workspaces that are created by TRE. |
 
-Alternatively, you can create the app registrations manually via the [Azure Portal](https://docs.microsoft.com/azure/active-directory/develop/quickstart-register-app). The requirements are listed below.
+Some of the applications require **admin consent** to allow them to validate users against the AAD. Check the Microsoft Docs on [Configure the admin consent workflow](https://docs.microsoft.com/en-us/azure/active-directory/manage-apps/configure-admin-consent-workflow) on how to request admin consent and handle admin consent requests.
 
-!!! note
-    Additional app registrations are required to run the E2E tests, and also to create workspaces - these are not configured by the `aad-app-reg.sh` script. Find information below on how to set these up.
+You can create these applications manually, but `/scripts/aad-app-reg.sh`  does the heavy lifting for you. Should you wish to create these manually via the [Azure Portal](https://docs.microsoft.com/azure/active-directory/develop/quickstart-register-app). The requirements are defined [below](#-Manual-Deployment).
```
Comment (daltskin) [inline docs/tre-admins/auth.md:20]: broken link , remove the "-" from (#Manual-Deployment)

File: docs/tre-admins/auth.md [hunk 4]
```diff
@@ -106,13 +138,32 @@ Redirect URLs:
 * `https://<TRE ID>.<Azure location>.cloudapp.azure.com/docs/oauth2-redirect`
 * `http://localhost:8000/docs/oauth2-redirect` - For local testing
 
+### Workspaces
+
+#### Authentication - Workspaces
+
+Same as [TRE API](#authentication---tre-api).
```
Comment (daltskin) [inline docs/tre-admins/auth.md:145]: broken link - should be (#authentication-tre-api)

**Review by daltskin — COMMENTED** [review]
LGTM

**Review by daltskin — APPROVED** [review]
LGTM

File: docs/tre-admins/auth.md [hunk 5]
```diff
@@ -1,55 +1,87 @@
...
 
-Some of the applications require **admin consent** to allow them to validate users against the AAD. Check the Microsoft Docs on [Configure the admin consent workflow](https://docs.microsoft.com/en-us/azure/active-directory/manage-apps/configure-admin-consent-workflow) on how to request admin consent and handle admin consent requests.
+## App registration script
 
-### App registration script
+### TRE API, Swagger and Automation Account
 
-The `/scripts/aad-app-reg.sh` script automatically sets up the app registrations with the required permissions to run Azure TRE. It will create and configure the two main app registrations: **TRE API** and **TRE Swagger UI**.
+The `/scripts/aad-app-reg.sh` script automatically sets up the app registrations with the required permissions to run Azure TRE. It will create and configure all of the app registrations depending on the parameters. You would typically run this script twice; once for the **TRE API** and once for each **Workspace**.
```
Comment (martinpeck) [inline docs/tre-admins/auth.md:26]: nit: Is "run this script twice" accurate? Would this be more accurate:

```
You would typically run this script multiple times; once for the **TRE API** and once for each **Workspace**.
```

Comment (martinpeck) [reply docs/tre-admins/auth.md:26]: This is a minor nit-pick, so please ignore for the purposes of this PR.

I've updated this branch with `main`

**Review by martinpeck — APPROVED** [review]

**Pull request merged**